src/device/pf_cmrpc.c

```diff
@@ -2536,16 +2552,120 @@ static int pf_cmrpc_dce_packet(
             }
 
             /* Handle the RPC request */
-            ret = pf_cmrpc_rpc_request(net, p_sess, req_pos, &rpc_req, *p_res_len, p_res, &res_pos);
+            p_sess->out_buf_len = 0;  /* Just to make sure */
+            ret = pf_cmrpc_rpc_request(net, p_sess, req_pos, &rpc_req, sizeof(p_sess->out_buffer), p_sess->out_buffer, &p_sess->out_buf_len);
+
+            if (res_pos + p_sess->out_buf_len < max_rsp_len)
+            {
+               /* NOT a fragmented answer - all fits into send buffer */
+               memcpy(&p_res[res_pos], p_sess->out_buffer, p_sess->out_buf_len);
+               p_sess->out_buf_sent_len = p_sess->out_buf_len;
+               res_pos += p_sess->out_buf_len;
+            }
+            else
+            {
+               LOG_DEBUG(PF_RPC_LOG, "CMRPC(%d): Send fragmented response, res_pos %u out_buf_len %u max is %u.\n", __LINE__, res_pos, p_sess->out_buf_len, max_rsp_len);
+
+               /* We ned to send a fragmented answer - Send the first fragment now */
+               memcpy(&p_res[res_pos], p_sess->out_buffer, max_rsp_len - res_pos);
+               p_sess->out_buf_sent_len = max_rsp_len - res_pos;
+               res_pos += max_rsp_len - res_pos;
 
-            /*
-             * FROM HERE ON:
-             * Do not use p_sess after this line.
-             * The RPC may have been a release. In that case the session does no longer exist.
-             */
+               /* Also set the fragment bit in the RPC response header */
+               rpc_res.flags.fragment = true;
+               rpc_res.flags.no_fack = false;
 
+               /* Re-write the header with the new info */
+               pf_put_dce_rpc_header(&rpc_res, max_rsp_len, p_res, &rpc_hdr_start_pos, &length_of_body_pos);
+
+               p_sess->out_fragment_nbr++; /* Number of the next fragment */
+            }
             /* Insert the real value of length_of_body in the rpc header */
-            pf_put_uint16(rpc_res.is_big_endian, (uint16_t)(res_pos - start_pos), *p_res_len, p_res, &length_of_body_pos);
+            pf_put_uint16(rpc_res.is_big_endian, (uint16_t)(res_pos - start_pos), max_rsp_len, p_res, &length_of_body_pos);
+            break;
+         case PF_RPC_PT_FRAG_ACK:
+            LOG_DEBUG(PF_RPC_LOG, "CMRPC(%d): Received fragment ACK.\n", __LINE__);
+            /* Handle fragment acknowledgment from the controller. */
+            if (p_sess->out_fragment_nbr > 0)
+            {
+               LOG_DEBUG(PF_RPC_LOG, "CMRPC(%d): Frag: ACK is valid.\n", __LINE__);
+               /* The fragment acknowledgment is valid (expected) */
+               if (p_sess->out_buf_len > p_sess->out_buf_sent_len)
+               {
+                  /* There is more to send - send it (or a part of it) */
+                  if (max_rsp_len > p_sess->ndr_data.args_maximum)
```

Comment:
This code looks similar to the code above. I dont see an actual error, but it should be checked together with the issue above.